src/codegen_call_io.cpp

```
@@ -111,510 +127,518 @@ static const CodeGen::NativeDispatchEntry io_table[] = {
+    if (key->getType() != cg.ptrTy_)
+        cg.codegenError("form_file() name must be str");
+    auto fn = cg.mod_->getOrInsertFunction("__ry_http_form_file", cg.fnTy_ptr_ptr_to_ptr_);
+    llvm::Value *result = cg.builder_.CreateCall(fn, {req, key}, "http_ffile");
+    llvm::Value *optResult = cg.wrapPtrAsOption(result, "http_ffile");
+    cg.type_meta_[CodeGen::TM_MapKey][optResult] = cg.ptrTy_;
+    cg.type_meta_[CodeGen::TM_MapValue][optResult] = cg.ptrTy_;
     return optResult;
 }
 
-llvm::Value *CodeGen::emitHttpListen(const CallExpr &e) {
+static llvm::Value *emitHttpListen(CodeGen &cg, const CallExpr &e) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)